### PR TITLE
Tank Mod cleanup and fix for blazemod

### PR DESCRIPTION
**Pull request by BorkBorkGoesTheCode**
Many redundant fields were removed and an artillery case abstract was added.

### CODE REVIEW

**Comment by hmstanley** [comment]
Are you going to clean up the linkage's as well?  Also, can we mainline the tread components so we might remove the blaze mod as a requirement?  Isn't that the only limitation since turrets were decided?

**Comment by hmstanley** [comment]
recoded. Sorry

**Comment by BorkBorkGoesTheCode** [comment]
>Are you going to clean up the linkage's as well? Also, can we mainline the tread components so we might remove the blaze mod as a requirement? Isn't that the only limitation since turrets were decided?

Don't know. Last I heard, they behave too much like wheels to be mainlined.

**Comment by BorkBorkGoesTheCode** [comment]
Linkages are broken in a different area, not this mod.

**Comment by DangerNoodle** [comment]
Last I checked, I could have sworn that mainline turrets still function as automated, while Blazemod turrets are manual. The mod also uses a few other parts for armoring, if I recall.

Also, two anomalies. First, self-propelled howitzer has a single heavy-duty board in the back extension section, when the identical sections next to it are quarterpanels. Second, the "military seat" seems to be pointless unless turrets still preclude installing cargo on the same tile. Even then, it seems odd that they have less durability than standard seats. Maybe they should be higher, and use a heavy duty frame instead of a seat for installing?

**Comment by mugling** [comment]
> Second, the "military seat" seems to be pointless unless turrets still preclude installing cargo on the same tile

Military seats are indeed pointless following the turret changes and can/should be deprecated

**Comment by DangerNoodle** [comment]
Potentially, yes. Either that or alter them, but it depends on whether a more durable seat has a useful niche for players. Then again, I do recall "death by car seat breaking while in motion" being referenced, though not as infamous as death via moose.

**Comment by mugling** [comment]
It only existed because you couldn't previously have turret and cargo on same tile. We don't need to find a new reason for it to exist - it can just go.

**Comment by DangerNoodle** [comment]
Understood.

**Comment by hmstanley** [comment]
are we gonna get this in a build soon?

**Comment by BorkBorkGoesTheCode** [comment]
Hi. Sorry for the delay, having to cope with continuing hacking and upheaval. I may expand this PR with some vehicle ideas.

**Comment by BorkBorkGoesTheCode** [comment]
Would anybody object if I added a 30mm antimateriel rifle?

**Comment by hmstanley** [comment]
Nah

**Comment by DangerNoodle** [comment]
I can see it being useful, since most of the tank ammunition has limited utility outside of the vehicle role. There is the man-portable TOW launcher, and that seems to be about it.

Peculiarly, the other weapons appear to be generic though. I assume the mod author had difficulty finding names and statistics of specific vehicle weapons?

**Comment by BorkBorkGoesTheCode** [comment]
My guess is that Chaosvolt kept them generic because it isn't possible to recreate irl tanks ingame yet.

**Comment by hmstanley** [comment]
Is this gonna get built today?

**Comment by DangerNoodle** [comment]
That might be another reason, given the main tanks displayed in the mod do not strongly resemble anything currently in use by the US armed forces that I know of, so it would be reasonable that new tank weapons could be fictional as well.

**Comment by BorkBorkGoesTheCode** [comment]
I'm going to stop adding features for now; the rifle and a Armored Recovery Vehicle I've been working on will be in the next PR.

**Comment by hmstanley** [comment]
Can you kick off a build or do you need to wait?

**Comment by BorkBorkGoesTheCode** [comment]
I lack merge rights.

**Comment by BorkBorkGoesTheCode** [comment]
Ready.

File: data/mods/Tanks/items.json [hunk 1]
```diff
@@ -1,27 +1,41 @@
 [
   {
+    "abstract": "artillery_casing",
+    "type": "GENERIC",
+    "category": "spare_parts",
+    "name": "A large artillery casing.",
+    "stackable": true,
+    "material": "steel",
```
Comment (mugling) [inline data/mods/Tanks/items.json]: Are they not brass? Hardly critical here. You could also consider `copy-from` the `abstract` casing

**Comment by mugling** [comment]
Multiple load errors

**Comment by BorkBorkGoesTheCode** [comment]
It looks like normal seats prevent installation of mounted weapons.

**Comment by mugling** [comment]
No, that shouldn't be the case

**Comment by BorkBorkGoesTheCode** [comment]
All seats with CARGO prevent turret installation.

**Comment by BorkBorkGoesTheCode** [comment]
The errors were caused by overwrite entries.

**Comment by DangerNoodle** [comment]
That suggests that the cargo flag still causes issues for turrets. Have you tested whether errors occur in these vehicles after fixing the overwrite issue, while using seats with cargo?

**Comment by BorkBorkGoesTheCode** [comment]
@DangerNoodle I have successfully installed mounted guns without errors after removing their cargo entries from blaze_override.json.

**Comment by DangerNoodle** [comment]
Ah, that would explain it.

File: data/mods/Tanks/items.json (hunk 1 above)
Comment (BorkBorkGoesTheCode) [inline data/mods/Tanks/items.json]: Artillery shells can be made of brass, steel, aluminum, and a wide variety of alloys.

**Comment by hmstanley** [comment]
Is this ever getting in?

**Comment by DangerNoodle** [comment]
It should, but the best that can be done is to be patient. Note that the last mod-focused pull request I had merged took nearly a month.

There is essentially a developer crisis going on due to the project being understaffed, and it is also why the undesirable idea of removing mods from the main game has floated about recently.

**Comment by BorkBorkGoesTheCode** [comment]
@chaosvolt do you object to any of these changes?

**Comment by BorkBorkGoesTheCode** [comment]
@Blaze-whatdoyoumeanitsalreadytaken do you object to any of the changes made to your mod?

**Comment by chaosvolt** [comment]
Hmm? Looks fine to me, and better than obsoleting yet another mod.

**Comment by mugling** [comment]
> do you object to any of these changes?

> do you object to any of the changes made to your mod?

Be aware nobody 'owns' any code shipped with the project

**Comment by mugling** [comment]
JSON load error:
```src/veh_type.cpp:454 [static void vpart_info::check()] vehicle part sickle has the TURRET flag, but is not made from a gun item```

Merging because it doesn't look significant and you have a dependent branch

**Pull request merged**